packages/net/tcp_connection_notify.pony

```diff
@@ -77,3 +77,22 @@ interface TCPConnectionNotify
     Called when the connection is closed.
     """
     None
+
+  fun ref throttled(conn: TCPConnection ref) =>
+    """
+    Called when the connection starts experiencing TCP backpressure. You should
+    respond to this by pausing additional calls to `write` and `writev` until
+    you are informed that pressure has been released. Failure to respond to
+    the `throttled` notification will result in outgoing data queuing in the
+    connection.
```

Comment:
maybe tack on something like "... and increasing memory usage" on the end to make the memory growth dangers a bit more explicit?